src/monitors/ModuleMonitor.v3

```diff
@@ -0,0 +1,273 @@
+// Copyright 2023 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+class ModuleMonitor(module: Module) {
```

Comment:
We should have a couple of lines of top-level documentation for this class. Ultimately I think this will be the main way monitors are implemented, so it's good to have an explanation of that at the class level.